engine/src/scheme.rs

```diff
@@ -12,10 +12,26 @@ use std::{
     cmp::{max, min},
     error::Error,
     fmt::{self, Debug, Display, Formatter},
+    iter::Iterator,
     ptr,
 };
 
-#[derive(PartialEq, Eq, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize)]
+#[serde(untagged)]
+/// Enum that descirbes a kind of index access operation
```

Comment:
Spelling mistake on `describe`

Reply:
I have written the doc comment for `FieldIndex` here, so can just use that! https://github.com/cloudflare/wirefilter/blob/grobinson/FW-1221/engine/src/scheme.rs#L20-L30